app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java

```diff
@@ -685,77 +685,243 @@ private double getSumDateOffHaveSalary() {
-                mRequestOff.getLeaveForMarriage())) {
-            mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(TypeOfDays.LEAVE_FOR_MARRIAGE);
-            mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getLeaveForMarriage());
-            mRequestDayOffTypesAttribute.setDestroy(FALSE);
-            mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+        mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+    }
+
+    private void setAttributeLeaveForChildMarriage() {
+        mRequestDayOffTypesAttribute = new OffRequest.RequestDayOffTypesAttribute();
+        mRequestDayOffTypesAttribute.setId(null);
+        mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(TypeOfDays.LEAVE_FOR_CHILD_MARRIAGE);
+        mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getLeaveForChildMarriage());
+        mRequestDayOffTypesAttribute.setDestroy(FALSE);
+        if (mRequestOff != null && (mRequestOff.getRequestDayOffTypes() != null
+                || mRequestOff.getRequestDayOffTypes().size() != 0)) {
+            for (int i = 0; i < mRequestOff.getRequestDayOffTypes().size(); i++) {
+                if (mRequestOff.getRequestDayOffTypes().get(i).getSpecialDayOffSettingId()
+                        == Integer.parseInt(
+                        RequestOffViewModel.TypeOfDays.LEAVE_FOR_CHILD_MARRIAGE)) {
+                    mRequestDayOffTypesAttribute.setId(
+                            mRequestOff.getRequestDayOffTypes().get(i).getId());
+                }
+            }
+            if (mRequestOff.getLeaveForChildMarriage() == null || "".equals(
+                    mRequestOff.getLeaveForChildMarriage()) || Constant.DEFAULT_DOUBLE_VALUE.equals(
+                    mRequestOff.getLeaveForChildMarriage())) {
+                mRequestDayOffTypesAttribute.setDestroy(TRUE);
+            }
         }
-        if (mRequestOff.getLeaveForChildMarriage() != null && !"".equals(
-                mRequestOff.getLeaveForChildMarriage())) {
-            mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(
-                    TypeOfDays.LEAVE_FOR_CHILD_MARRIAGE);
-            mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getLeaveForChildMarriage());
-            mRequestDayOffTypesAttribute.setDestroy(FALSE);
-            mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+        mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+    }
+
+    private void setAttributeMaternityLeave() {
+        mRequestDayOffTypesAttribute = new OffRequest.RequestDayOffTypesAttribute();
+        mRequestDayOffTypesAttribute.setId(null);
+        mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(TypeOfDays.MATERNTY_LEAVE);
+        mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getMaternityLeave());
+        mRequestDayOffTypesAttribute.setDestroy(FALSE);
+        if (mRequestOff != null && (mRequestOff.getRequestDayOffTypes() != null
```

Comment:
ERROR: [BrokenNullCheck] Method call on object which may be null
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/basic.html#BrokenNullCheck